contracts/smart-contract-wallet/SmartAccount.sol

```diff
@@ -372,10 +381,11 @@ contract SmartAccount is
     ) external returns (uint256) {
         uint256 startGas = gasleft();
         // We don't provide an error message here, as we use it to return the estimate
-        require(execute(to, value, data, operation, gasleft()));
-        uint256 requiredGas = startGas - gasleft();
+        if(!execute(to, value, data, operation, gasleft())) revert ExecutionFailed();
         // Convert response to string and return via error message
-        revert(string(abi.encodePacked(requiredGas)));
+        unchecked { 
```

Comment:
something similar was in the QA. taking a note so it helps with resolving conflicts.